NDTensors/src/lib/Sectors/src/abstractcategory.jl

```diff
@@ -1,45 +1,129 @@
+function fusion_rule(::SymmetryStyle, c1::C, c2::C) where {C<:AbstractCategory}
+  degen, labels = label_fusion_rule(C, category_label(c1), category_label(c2))
+  return GradedAxes.gradedrange(LabelledNumbers.labelled.(degen, C.(labels)))
 end
 
-function trivial(category_type::Type{<:AbstractCategory})
-  return error("`trivial` not defined for type $(category_type).")
+# abelian case: return Category
+function fusion_rule(::AbelianGroup, c1::C, c2::C) where {C<:AbstractCategory}
+  return C(label_fusion_rule(C, category_label(c1), category_label(c2)))
 end
 
-istrivial(c::AbstractCategory) = (c == trivial(typeof(c)))
+function fusion_rule(
+  ::EmptyCategory, l1::LabelledNumbers.LabelledInteger, l2::LabelledNumbers.LabelledInteger
+)
+  return LabelledNumbers.labelled(l1 * l2, sector())
+end
+
+function label_fusion_rule(category_type::Type{<:AbstractCategory}, ::Any, ::Any)
+  return error("`label_fusion_rule` not defined for type $(category_type).")
+end
 
-function dual(category_type::Type{<:AbstractCategory})
-  return error("`dual` not defined for type $(category_type).")
+# ================================  GradedAxes interface  ==================================
+# tensor_product interface
+function GradedAxes.fuse_blocklengths(
+  l1::LabelledNumbers.LabelledInteger{<:Integer,<:Sectors.AbstractCategory},
+  l2::LabelledNumbers.LabelledInteger{<:Integer,<:Sectors.AbstractCategory},
+)
+  return GradedAxes.fuse_blocklengths(
+    combine_styles(SymmetryStyle(l1), SymmetryStyle(l2)), l1, l2
+  )
 end
 
-Base.isless(c1::AbstractCategory, c2::AbstractCategory) = isless(label(c1), label(c2))
+function GradedAxes.fuse_blocklengths(
+  ::SymmetryStyle, l1::LabelledNumbers.LabelledInteger, l2::LabelledNumbers.LabelledInteger
+)
+  fused = LabelledNumbers.label(l1) ⊗ LabelledNumbers.label(l2)
+  v =
+    LabelledNumbers.labelled.(
+      l1 * l2 .* BlockArrays.blocklengths(fused), GradedAxes.blocklabels(fused)
+    )
+  return GradedAxes.gradedrange(v)
+end
+
+function GradedAxes.fuse_blocklengths(
+  ::AbelianGroup, l1::LabelledNumbers.LabelledInteger, l2::LabelledNumbers.LabelledInteger
+)
+  fused = LabelledNumbers.label(l1) ⊗ LabelledNumbers.label(l2)
+  return LabelledNumbers.labelled(l1 * l2, fused)
+end
+
+function GradedAxes.fuse_blocklengths(
+  ::EmptyCategory, l1::LabelledNumbers.LabelledInteger, l2::LabelledNumbers.LabelledInteger
+)
+  return LabelledNumbers.labelled(l1 * l2, sector())
+end
+
+# cast to range
+to_graded_axis(c::AbstractCategory) = to_graded_axis(LabelledNumbers.labelled(1, c))
+to_graded_axis(l::LabelledNumbers.LabelledInteger) = GradedAxes.gradedrange([l])
+to_graded_axis(g::AbstractUnitRange) = g
+
+# allow to fuse a category with a GradedUnitRange
+function GradedAxes.tensor_product(c::AbstractCategory, g::AbstractUnitRange)
+  return GradedAxes.tensor_product(to_graded_axis(c), g)
+end
+
+function GradedAxes.tensor_product(g::AbstractUnitRange, c::AbstractCategory)
+  return GradedAxes.tensor_product(c, g)
```

Comment:
It doesn't seem like a good idea to switch the order like this, since that seems to assume commutativity.

Reply:
We specifically ruled non-commutative fusion rings out of the scope of supported features, so I think this is fine.

Reply:
I see, anyway I would prefer to define this as:
```julia
function GradedAxes.tensor_product(g::AbstractUnitRange, c::AbstractCategory)
  return GradedAxes.tensor_product(g, to_graded_axis(c))
end
```
It's not very complicated and is more general in case we want to generalize at some point.